src/frontend/komponenter/Fagsak/Behandlingsresultat/Kompetanse/KompetanseTabellRadEndre.tsx

```diff
@@ -0,0 +1,541 @@
+import React, { useState } from 'react';
+
+import styled from 'styled-components';
+
+import navFarger from 'nav-frontend-core';
+import { SkjemaGruppe } from 'nav-frontend-skjema';
+import { Element } from 'nav-frontend-typografi';
+
+import { Delete } from '@navikt/ds-icons';
+import { Alert } from '@navikt/ds-react';
+import {
+    FamilieKnapp,
+    FamilieReactSelect,
+    FamilieSelect,
+    type OptionType,
+} from '@navikt/familie-form-elements';
+import { type FeltState, Valideringsstatus } from '@navikt/familie-skjema';
+import { type Ressurs, RessursStatus } from '@navikt/familie-typer';
+import type { Country } from '@navikt/land-verktoy';
+
+import { useBehandling } from '../../../../context/behandlingContext/BehandlingContext';
+import { KompetanseSubmit, useKompetanse } from '../../../../context/Kompetanse/KompetanseContext';
+import { validerKompetanse } from '../../../../context/Kompetanse/valideringKompetanse';
+import type { IBehandling } from '../../../../typer/behandling';
+import {
+    AnnenForelderAktivitet,
+    annenForelderAktiviteter,
+    type IKompetanse,
+    SøkerAktivitet,
+    søkerAktiviteter,
+    KompetanseResultat,
+    kompetanseResultater,
+} from '../../../../typer/kompetanse';
+import { nyYearMonthPeriode } from '../../../../utils/kalender';
+import IkonKnapp, { IkonPosisjon } from '../../../Felleskomponenter/IkonKnapp/IkonKnapp';
+import MånedÅrVelger from '../../../Felleskomponenter/MånedÅrInput/MånedÅrVelger';
+import FamilieLandvelger from './FamilieLandvelger';
+import { kompetanseFeilmeldingId, kompetansePeriodeFeilmeldingId } from './KompetanseSkjema';
+
+const Container = styled.div`
+    max-width: 30rem;
+    border-left: 0.0625rem solid ${navFarger.navOransje};
+    padding-left: 2rem;
+`;
+
+const StyledLegend = styled.legend`
+    && {
+        display: flex;
+        margin-bottom: 0;
+    }
+`;
+
+const FlexDiv = styled.div`
+    width: 28rem;
+    display: flex;
+    justify-content: space-between;
+
+    div {
+        z-index: 0;
+    }
+
+    div div.skjemaelement {
+        margin-bottom: 0rem;
+    }
+`;
+
+const Knapperad = styled.div`
+    display: flex;
+    justify-content: space-between;
+    margin: 1rem 0;
+    margin-top: 2rem;
+`;
+
+interface IProps {
+    redigerbartKompetanse: FeltState<IKompetanse>;
+    settRedigerbartKompetanse: (redigerbartKompetanse: FeltState<IKompetanse>) => void;
+    tilgjengeligeBarn: OptionType[];
+    visFeilmeldinger: boolean;
+    toggleForm: (visAlert: boolean) => void;
+    settEkspandertKompetanse: (ekspandertKompetanse: boolean) => void;
+}
+
+const KompetanseTabellRadEndre: React.FC<IProps> = ({
```

Comment:
Klarer vi å splitte opp denne komponenten litt? 541 linjer ble mye for meg å ta inn 😅